src/core/ddsi/src/ddsi_plist.c

```diff
@@ -3280,6 +3278,9 @@ void ddsi_xqos_init_default_reader (dds_qos_t *xqos)
   xqos->type_consistency.ignore_member_names = false;
   xqos->type_consistency.prevent_type_widening = false;
   xqos->type_consistency.force_type_validation = false;
+
+  xqos->present |= QP_LOCATOR_MASK;
+  xqos->ignore_locator_type = 0;
```

Comment:
I agree with the decision to add this here; but then in `dds_create_reader` and `dds_create_writer`, it must be present already and there is no need to add `QP_LOCATOR_MASK` another time.

Reply:
You are correct in this.